doc/pit/Platform_Integration_Test_high_level_design.md

```diff
@@ -0,0 +1,217 @@
+
+Test case list generator is used to generate a list of test cases, it parses users' input options, product specific platform configuration, and generate test case list from all test case database. This is used in the daily regression test and delivery test scenario.
+
+Each test case has a list of tags, denoting type, stage, etc. Current tag for stage includes: production, development, delivery. Tags for type include: auto, manual, utility. Other tags can be added, to provide extra information so that they can be grouped together for new scenario.
+
+### 3.4.2 Manual test options
+
+Alternatively, user can specify which case or a list of test cases to run. Optionally, users are able to specify how many times to run the cases. This is used in development stage for developers to debug/test a single feature, or be used in stress test scenario.
+
+### 3.4.3 Test result generator
+
+Test result is generated per case. There are 2 result formats:
+
+  1) Formatted strings, contains test case name, ID, result, type, etc. This standardized output is used in manufacturing process, so that output strings can be mapped to hardware/software/firmware/configuration fault easily.
+  2) Structured result, per current design, we're using JSON format, as it's commonly used and can be interfaced with users' automated acceptance checking system in the product delivery process.
+
+Besides test case result, a collective result is also generated, as a summary of the whole test. Similar to per test case result, it composes of standardized output string and structured data format (JSON).
+
+Figure 3-3 shows an example of final test result JSON.
+![](images/PIT_final_test_result.png)
+Figure 3-3
+
+### 3.4.4 Test case logic
+
+Figure 3-4 shows the test case logic and interaction
+![](images/PIT_test_case_logic.png)
+Figure 3-4
+
++Each test case has a configuration file, containing information such as test case name, description, type, and tag list. An example test case configuration file shown as follow:
++Path: $(PIT_ROOT_DIR)/cases/$(TEST_CASE_NAME)/config.json
++Content:
++{
++    "name": "fan-test",
++    "description": "Check fan status",
++    "type": "auto",
++    "tags": ["manufacture", "delivery"]
++}
++Tags shown above "manufacture" denotes this case should be included in Manufaturing-test, "delivery" denotes this case should be included in Accceptance-test. Also, we may expanded tags if it's neccessary. Tags are defined as strings.
++
++Each platform has a platform configuration file(divided into 2 files: platform_config.json, case_config.json), containing platform specific (product specific) information such as CPU architecture, boot loader, CPU error monitor capability, peripheral related information (e.g. FAN number, PSU control options, etc.). An example of configuration file is shown as follow:
+Path: $(PIT_SRC_DIR)/config/platform/$(PLATFORM_NAME)/platform_config.json, where $PLATFORM_NAME is 'x86_64-alibaba_as14-40d-cl-r0' in this case
```

Comment:
Just checking, do you have these file names reversed? This looks more like a case_config.json just from the naming.

Reply:
For file platform_config.json, we currently only defines 1 key, i.e  "test_cases", to specify a default test cases list for "-d" option.

The initial purpose of this file is to let users to customize the product(platform). E.g user may want to decide what cases should be covered by acceptance test. One vendor may sell 1 product to multiple users, and different users may want a different checklist.